src/org/omegat/core/search/Searcher.java

```diff
@@ -662,6 +681,55 @@ public boolean searchString(String origText, boolean collapseResults) {
         return true;
     }
 
+    // Interpret case changes in regex. Must be called _AFTER_ variables replacement!
+    private static String replaceCase(String txt) {
+        if (!txt.startsWith("\\")) {
+            int idx = txt.indexOf("\\");
+            if (idx == -1) {
+                return txt;
+            } else {
+                return txt.substring(0, idx) + replaceCase(txt.substring(idx));
+            }
+        }
+        // Double symbols are longer, so they must be treated first
+        if (txt.startsWith("\\u\\L")) {
+            return txt.substring(4, 5).toUpperCase() + replaceCase("\\L" + txt.substring(5));
```

Comment:
Correct (I was alerted by Didier just after I sent the patch)
One point which may cause problems, however: 
logically, the target language should be used;
but in OmegaT, the replace screen can be used to fill untranslated segments, in which case it "replaces" based on source language. In other terms, there is a use case where source language should be used instead of target one.
What do you think should be done?

Reply:
Unfortunately it seems even worse than that: `replaceCase` is used in `searchString`, and we call this against lots of different text that could be in any language, such as notes and external files.

Ideally we can handle this by supplying the language/locale as a parameter to `searchString`. I think it's fair to assume that notes and external files should be in the source language.

Reply:
That should be a good reason why to split this big "Searcher" class into specific ones, linked by inheritance: then "searchString" would be overridden and only the version specific to replacements would make calls to replaceCase.